src/jqLite.js

```diff
@@ -671,7 +671,10 @@ forEach({
       }
       return element.value;
     }
-    element.value = value;
+    // Workaround for Firefox validation #12102.
+    if (element.value != value) {
```

Comment:
The rules for transforming to string are not the same as the rules for `!=` comparison.
Maybe you should compare `element.value` to `String(value)` or something ?